src/NuGet.Core/NuGet.Packaging/Signing/Utility/SignatureUtility.cs

```diff
@@ -583,7 +583,9 @@ private static bool IsMatch(X509Certificate2 certificate, EssCertId essCertId)
 
             byte[] actualHash;
 
+#pragma warning disable CS0618 // Type or member is obsolete
             using (var hashAlgorithm = CryptoHashUtility.GetSha1HashProvider())
+#pragma warning restore CS0618 // Type or member is obsolete
```

Comment:
The obsolete message says to use `GetSha1HashProvider`, so this product code should use that instead, or this method should also be marked obsolete.

Reply:
The obsolete message says to call `GetHashAlgorithm` instead of `GetSha1HashProvider`. `GetHashAlgorithm` supports only [`SHA256 or SHA384 or SHA512`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/HashAlgorithmName.cs#L8-L11) hash algorithms and can't be invoked here because `SHA1` is not supported. That's why I thought of adding an `Obsolete` attribute to [`GetSha1HashProvider`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/CryptoHashUtility.cs#L206) method and add `#pragma warning` to all the existing callers so that build won't break and new callers will be notified if this method is invoked. New callers can also add `#pragma warning` if `SHA1` is required as per design.

> or this method should also be marked obsolete.

TIL, marking method as `Obsolete` is another way to avoid disabling warnings in the source. In this case, I am worried that the entire call stack has to be marked `Obsolete`. May be my idea to mark `CryptoHashUtility.GetSha1HashProvider` as `Obsolete` is bad.